File: dayfour/code.java
import java.io.File;
import java.io.FileNotFoundException;
import java.util.Scanner;

public class advent {

  public static void main(String[] args) {
    System.out.println(one());
    System.out.println(two());

  }

  private static int one() {
    File f = new File("input.txt");
    Scanner scnr = null;
    try {
      scnr = new Scanner(f);
    } catch (FileNotFoundException e) {
      e.printStackTrace();
    }

    int total = 0;

    while (scnr.hasNextLine()) {
      String currLine = scnr.nextLine().strip();
      String[] parts = currLine.split(",");
      String[] firstRange = parts[0].split("-");
      int one = Integer.parseInt(firstRange[0]);
      int two = Integer.parseInt(firstRange[1]);
      String[] secondRange = parts[1].split("-");
      int three = Integer.parseInt(secondRange[0]);
      int four = Integer.parseInt(secondRange[1]);

      if ((one <= three && two >= four) || (three <= one && four >= two))
        total += 1;

    }

    return total;
  }

  private static int two() {
    File f = new File("input.txt");
    Scanner scnr = null;
    try {
      scnr = new Scanner(f);
    } catch (FileNotFoundException e) {
      e.printStackTrace();
    }

    int total = 0;

    while (scnr.hasNextLine()) {
      String currLine = scnr.nextLine().strip();
      String[] parts = currLine.split(",");
      String[] firstRange = parts[0].split("-");
      int one = Integer.parseInt(firstRange[0]);
      int two = Integer.parseInt(firstRange[1]);
      String[] secondRange = parts[1].split("-");
      int three = Integer.parseInt(secondRange[0]);
      int four = Integer.parseInt(secondRange[1]);
      int min = Math.max(one,three);
      int max = Math.min(two,four);
      
      if(max - min >= 0)
        total += 1;

    }

    return total;
  }
}

File: dayfour/code.py
def one():
    file = open('input.txt', 'r')
    lines = file.readlines()
    total = 0
    for line in lines:
        line = line.strip()
        both = line.split(",")
        first_range = both[0]
        first_range = first_range.split("-")
        num_1 = int(first_range[0])
        num_2 = int(first_range[1])
        second_range = both[1]
        second_range = second_range.split("-")
        num_3 = int(second_range[0])
        num_4 = int(second_range[1])

        if (num_1 <= num_3 and num_2 >= num_4) or (num_3 <= num_1 and num_4 >= num_2):
            total += 1

    return total


def two():
    file = open('input.txt', 'r')
    lines = file.readlines()
    total = 0
    for line in lines:
        line = line.strip()
        both = line.split(",")
        first_range = both[0]
        first_range = first_range.split("-")
        num_1 = int(first_range[0])
        num_2 = int(first_range[1])
        second_range = both[1]
        second_range = second_range.split("-")
        num_3 = int(second_range[0])
        num_4 = int(second_range[1])

        xy = range(max(num_1, num_3), min(num_2 + 1, num_4 + 1))
        if len(xy) > 0:
            total += 1

    return total


if __name__ == '__main__':
    print(one())
    print(two())
